src/compiler/scala/tools/nsc/transform/CleanUp.scala

```diff
@@ -501,6 +501,65 @@ abstract class CleanUp extends Statics with Transform with ast.TreeDSL {
         reducingTransformListApply(rest.elems.length) {
           super.transform(localTyper.typedPos(tree.pos)(consed))
         }
+      //methods on Double
+      //new Predef.doubleToDouble(x).isNaN() -> java.lang.Double.isNaN(x)
+      //new Predef.doubleToDouble(x).isInfinite() -> java.lang.Double.isInfinity(x)
+      //methods on Float
+      //new Predef.float2Float(x).isNaN() -> java.lang.Double.isNaN(x)
+      //new Predef.float2Float(x).isInfinite() -> java.lang.Double.isInfinity(x)
+
+      //methods on Number
+      //new Predef.<convert>(x).byteValue() -> x.toByte()
+      //new Predef.<convert>(x).shortValue() -> x.toShort()
+      //new Predef.<convert>(x).intValue() -> x.toInt()
+      //new Predef.<convert>(x).longValue() -> x.toLong()
+      //new Predef.<convert>(x).floatValue() -> x.toFloat()
+      //new Predef.<convert>(x).doubleValue() -> x.toDouble()
+      //
+      // for each of the conversions
+      // double2Double
+      // float2Float
+      // byte2Byte
+      // short2Short
+      // char2Character
+      // int2Integer
+      // long2Long
+      // boolean2Boolean
+      //
+      case Apply(Select(Apply(boxing, params), methodName), Nil)
+        if currentRun.runDefinitions.PreDef_primitives2Primitives.contains(boxing.symbol) &&
+          params.size == 1 &&
+          allFloatingMethodsToRewrite.contains(methodName) =>
```

Comment:
The rewrite drops the qualifier of the boxing method. So for example,

```
scala> {println("hi"); Predef}.int2Integer(1).doubleValue
```

is re-written and the side effect is dropped.
```suggestion
      case Apply(Select(Apply(boxing @ Select(qual, _), params), methodName), Nil)
        if currentRun.runDefinitions.PreDef_primitives2Primitives.contains(boxing.symbol) &&
          params.size == 1 &&
          allFloatingMethodsToRewrite.contains(methodName) &&
          treeInfo.isExprSafeToInline(qual) =>
```